src/com/facebook/buck/apple/toolchain/impl/AppleToolchainDiscovery.java

```diff
@@ -73,7 +76,12 @@ private AppleToolchainDiscovery() {}
 
       try (DirectoryStream<Path> toolchainStream =
           Files.newDirectoryStream(toolchains, "*.xctoolchain")) {
-        for (Path toolchainPath : toolchainStream) {
+
+        List<Path> sortedToolchainPaths = new ArrayList<>();
+        toolchainStream.forEach(sortedToolchainPaths::add);
+        sortedToolchainPaths.sort(Comparator.comparing(Path::toString));
+
+        for (Path toolchainPath : sortedToolchainPaths) {
```

Comment:
Can you replace it with `ImmutableSortedSet.copyOf(toolchainStream)` and remove the sorting?